src/snapshot_tests/tests.rs

```diff
@@ -18,10 +20,143 @@ pub fn snapshot_tests() -> Vec<TestCase> {
     tests.append(&mut text_snapshot_tests());
     tests.append(&mut tiles_snapshot_tests());
     tests.append(&mut rescaler_snapshot_tests());
+    tests.append(&mut shader_tests());
     tests
 }
 
-pub fn rescaler_snapshot_tests() -> Vec<TestCase> {
+fn shader_tests() -> Vec<TestCase> {
+    let input1 = TestInput::new(1);
+    let input2 = TestInput::new(2);
+    let input3 = TestInput::new(3);
+    let input4 = TestInput::new(4);
+    let input5 = TestInput::new(5);
+
+    lazy_static! {
+        static ref PLANE_ID_SHADER: String = json!({
+            "type": "register",
+            "entity_type": "shader",
+            "shader_id": "base_params_plane_id",
+            "source": include_str!("../../snapshot_tests/shader/base_params_plane_id.wgsl")
+        })
+        .to_string();
+        static ref TIME_SHADER: String = json!({
+            "type": "register",
+            "entity_type": "shader",
+            "shader_id": "base_params_time",
+            "source": include_str!("../../snapshot_tests/shader/base_params_time.wgsl")
+        })
+        .to_string();
+        static ref TEXTURE_COUNT_SHADER: String = json!({
+            "type": "register",
+            "entity_type": "shader",
+            "shader_id": "base_params_texture_count",
+            "source": include_str!("../../snapshot_tests/shader/base_params_texture_count.wgsl")
+        })
+        .to_string();
+        static ref OUTPUT_RESOLUTION_SHADER: String = json!({
+            "type": "register",
+            "entity_type": "shader",
+            "shader_id": "base_params_output_resolution",
+            "source": include_str!("../../snapshot_tests/shader/base_params_output_resolution.wgsl")
+        })
+        .to_string();
+    };
+
+    Vec::from([
+        TestCase {
+            name: "shader/base_params_plane_id_no_inputs",
+            outputs: Outputs::Scene(vec![(
+                include_str!(
+                    "../../snapshot_tests/shader/base_params_plane_id_no_inputs.scene.json"
+                ),
+                DEFAULT_RESOLUTION,
+            )]),
+            renderers: vec![&PLANE_ID_SHADER],
+            ..Default::default()
+        },
+        TestCase {
+            name: "shader/base_params_plane_id_5_inputs",
+            outputs: Outputs::Scene(vec![(
+                include_str!(
+                    "../../snapshot_tests/shader/base_params_plane_id_5_inputs.scene.json"
+                ),
+                DEFAULT_RESOLUTION,
+            )]),
+            renderers: vec![&PLANE_ID_SHADER],
+            inputs: vec![
+                input1.clone(),
+                input2.clone(),
+                input3.clone(),
+                input4.clone(),
+                input5.clone(),
+            ],
+            ..Default::default()
+        },
+        TestCase {
+            name: "shader/base_params_time_5s",
```

Comment:
remove that 5s, it's confusing, if you want you can add here in a comment or in wgsl code comment that entire animation takes 5 seconds